### ISSUE TITLE
[Card - Pictogram] screen reader is not announcing the role. User would not know it is an interactive element

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#7900]
### Description

Issue:
On tab navigation entire card is getting focused and users are able to activate it, however screen reader is not announcing the role. Hence screen reader user would not know it is an interactive element, thus may not activate it.

Expected: 
Anchor tag should be placed in a logical spot in the card with an appropriate label and the same should be identified by the screen reader and receive keyboard focus in tab/shift tab navigation

### Component(s) impacted

.

### Browser

Firefox

### Carbon for IBM.com version

1.28

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

Web component

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

NA

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
